yusuke-sforzando/q-lako
# Show details of the search result and register Airtable

## Describe the PR

To close #5  .
Show details of the search result.

## Screenshots

### registration.html

![Screen Shot 2020-11-28 at 16 13 37](https://user-images.githubusercontent.com/67737076/100496100-b6feba00-3194-11eb-8e65-d609d0edb2c9.png)

## Detail of the change

1. main.py
    - Receive `asin` from `GET /search`
    - Run`amazon_api_client.search_products(keywords=***)`
    - Format response data from Amazon PA-API to show as details
        - → dataclass `Asset`
    - Run `return render_template("registration.html",  **context_dict)`
1. registration.html
    - Show data with Jinja template as shown below

```html
<div>
  <h1>登録内容詳細</h1>
  <image src="VALUE_FROM_AMAZON_PA-API" />
  <p>商品名/書籍名: TITLE_FROM_AMAZON_PA-API</p>
  …
  <p>登録者: VALUE_OF_REGISTRANT_NAME</p>
</div>
```

## Additional context

`registered_at` not need to display because it is automatically filled at posted.


## To reproduce

Steps to check the behavior:

1. `python main.py`
1. Access to `http://0.0.0.0:8888/` on browser
1. Enter any keyword, ASIN or ISBN code and press the search button
1. Select any item from the list and press the Select button
1. Make sure the detailed information of the item appears as a screenshot

## Additional context

None.
